ompi/dpm/dpm.c

```diff
@@ -1974,6 +1975,46 @@ static void set_handler_default(int sig)
     sigaction(sig, &act, (struct sigaction *)0);
 }
 
+static char *find_prte(void)
+{
+    char *filename = NULL;
+#if !OMPI_USING_INTERNAL_PRRTE
+    char *prrte_prefix = NULL;
+#endif
+
+    /* 1) Did the user tell us exactly where to find prte? */
+    filename = getenv("OMPI_PRTERUN");
+    if (NULL != filename) {
+        return filename;
```

Comment:
@hppritcha Ah, I see.  So you're suggesting we need this?

```suggestion
        return strdup(filename);
```

Reply:
@rhc54 I took the liberty of updating the PR to include the `strdup()`.